frontend/ui/lib/widgets/autocomplete_text_field_widget.dart

```diff
@@ -1,2 +1,100 @@
-export 'package:tripmeout/widgets/autocomplete_text_field_widget_stub.dart'
-    if (dart.library.html) 'package:tripmeout/widgets/autocomplete_text_field_widget_html.dart';
+import 'dart:async';
+import 'package:flutter/material.dart';
+import 'package:tripmeout/model/place_visit.dart';
+import 'package:flutter_typeahead/flutter_typeahead.dart';
+import 'package:tripmeout/services/places_services.dart';
+import 'package:flutter/src/widgets/basic.dart' as basic;
+
+class MapsApiPlacesTextFieldWidget extends StatefulWidget {
+  final List<String> allowedTypes;
+  final PlacesApiServices placesApiServices;
+  MapsApiPlacesTextFieldWidget(this.allowedTypes, this.placesApiServices);
+
+  @override
+  _MapsApiPlacesTextFieldState createState() =>
+      _MapsApiPlacesTextFieldState();
+}
+
+class _MapsApiPlacesTextFieldState extends State<MapsApiPlacesTextFieldWidget> {
+  final TextEditingController _typeAheadController = TextEditingController();
+  PlacesApiServices get placesApiServices => widget.placesApiServices;
+  List<String> get allowedTypes => widget.allowedTypes;
+
+  _MapsApiPlacesTextFieldState();
+
+  String placeId = "";
+  Future<List<String>> image;
+  bool show = false;
+
+  void showImage(String placeId, bool show) {
+    setState(() {
+      this.placeId = placeId;
+      this.show = show;
+      if (show) {
+        this.image = placesApiServices.getPhotos(this.placeId);
+      } else {
+          this.image = null;
+      }
+
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        basic.Padding(
+          padding: const EdgeInsets.all(25.0),
+          child: TypeAheadField<PlaceVisit>(
+            textFieldConfiguration: TextFieldConfiguration(
+                autofocus: true,
+                decoration: InputDecoration(
+                  border: OutlineInputBorder(),
+                  labelText: 'Enter your Destination',
+                ),
+                controller: this._typeAheadController),
+            suggestionsCallback: (pattern) =>
+                placesApiServices.getAutocomplete(pattern, allowedTypes),
+            itemBuilder: (context, suggestion) {
+              return ListTile(
+                title: Text(suggestion.name),
+              );
+            },
+            onSuggestionSelected: (suggestion) {
+              this._typeAheadController.text = suggestion.name;
+              showImage(suggestion.placesApiPlaceId, true);
+            },
+            noItemsFoundBuilder: (BuildContext context) => Text('Please enter a city'),
+          ),
+        ),
+        FutureBuilder<List<String>>(
+            future: image,
+            builder: (BuildContext context, AsyncSnapshot<List<String>> snapshot) {
+              if (snapshot.hasData) {
+                List<Widget> imageWidgets = imagesToContainer(snapshot.data);
```

Comment:
Still looking through the rest of this CL, but the fact that the text field is the one rendering the images is just weird.

I would very much prefer that the autocomplete field be a very small piece of functionality that just takes care of auto complete stuff. It can probably take a callback for when a place is selected.